### PR TITLE
feat: Data.BigNum: from_num method implement

### PR DESCRIPTION
**Pull request by tsuyoshicho**
Currently Data.BigNum.from_num() call from_string with stringnized args.

This PR is replace to legal generation from num code.

### CODE REVIEW

**Comment by ujihisa** [comment]
What's legal generation?

File: autoload/vital/__vital__/Data/BigNum.vim [hunk 1]
```diff
@@ -11,18 +20,43 @@ lockvar! s:_NODE_MAX_DIGIT
...
+  return s:_fix_form(l:bignum)
 endfunction
 
 function! s:from_string(str) abort
-  if s:_is_number(a:str) != 1
-    call s:_throw('is not number: '.a:str)
+  if !s:_is_number(a:str)
+    call s:_throw('is not string or non-number string: ' . string(a:str))
   endif
+
```
Comment (ujihisa) [inline autoload/vital/__vital__/Data/BigNum.vim:59]: Don't these need to add/modify existing test codes?

**Review by ujihisa — APPROVED** [review]

File: autoload/vital/__vital__/Data/BigNum.vim (hunk 1 above)
Comment (tsuyoshicho) [reply autoload/vital/__vital__/Data/BigNum.vim:59]: I see.

Invalid data test add.

Comment (tsuyoshicho) [reply autoload/vital/__vital__/Data/BigNum.vim:59]: But In old test use from_num any patterns, do not need adding with normal value.

**Comment by tsuyoshicho** [comment]
Test success BigNum, failure only MacVim timing issue.

**Review by ujihisa — APPROVED** [review]

**Comment by ujihisa** [comment]
What's legal generation?

**Comment by tsuyoshicho** [comment]
Thanks a lot.

**Comment by tsuyoshicho** [comment]
X legal -> O direct... fixit desc.

**Pull request merged**